cpp/include/raft/linalg/transpose.cuh

```diff
@@ -55,7 +56,44 @@ void transpose(math_t* inout, int n, cudaStream_t stream)
   detail::transpose(inout, n, stream);
 }
 
+/**
+ * @brief Transpose a matrix. The output has same layout policy as the input.
+ *
+ * @tparam T Data type of input matrix element.
+ * @tparam LayoutPolicy Layout type of the input matrix. When layout is strided, it can
+ *                      be a submatrix of a larger matrix. Arbitrary stride is not supported.
+ * @tparam AccessorPolicy Accessor for the input and output, must be valid accessor on
+ *                        device.
+ *
+ * @param[in]  handle raft handle for managing expensive cuda resources.
+ * @param[in]  in     Input matrix.
+ * @param[out] out    Output matirx, storage is pre-allocated by caller.
+ */
+template <typename T, typename LayoutPolicy, typename AccessorPolicy>
+auto transpose(handle_t const& handle,
+               mdspan<T, ::raft::detail::matrix_extent, LayoutPolicy, AccessorPolicy> in,
```

Comment:
These have been brought into the `raft` namespace https://github.com/rapidsai/raft/blob/af4f35cc0608730f090122dc793fd8cc8cccc9da/cpp/include/raft/core/mdarray.hpp#L64

Reply:
I'd also like to see `mdspan` being used as `raft::mdspan`

Reply:
Fixed, thank you for the suggestions.